src/module/ovs/event.rs

```diff
@@ -323,15 +325,84 @@ impl EventFmt for ActionEvent {
-
         if self.recirc_id != 0 {
             write!(f, " recirc_id {}", self.recirc_id)?;
         }
 
+        match &self.action {
+            OvsAction::Unspecified => write!(f, " (unspecified)")?,
+            OvsAction::Output(a) => write!(f, " oport {}", a.port)?,
+            OvsAction::Userspace => write!(f, " userspace")?,
+            OvsAction::Set => write!(f, " tunnel_set")?,
+            OvsAction::PushVlan => write!(f, " push_vlan")?,
+            OvsAction::PopVlan => write!(f, " pop_vlan")?,
+            OvsAction::Sample => write!(f, " sample")?,
+            OvsAction::Recirc(a) => write!(f, " recirc {}", a.id)?,
+            OvsAction::Hash => write!(f, " hash")?,
+            OvsAction::PushMpls => write!(f, " push_mpls")?,
+            OvsAction::PopMpls => write!(f, " pop_mpls")?,
+            OvsAction::SetMasked => write!(f, " set_masked")?,
+            OvsAction::Ct(ct) => {
+                let mut args = Vec::new();
+                args.push(format!("zone={}", ct.zone_id));
+                if let Some(nat) = &ct.nat {
+                    let mut nat_args = match nat.dir {
+                        NatDirection::Src => "src".to_string(),
+                        NatDirection::Dst => "dst".to_string(),
+                    };
+                    if ct.flags & R_OVS_CT_NAT_RANGE_MAP_IPS != 0 {
+                        if let (Some(min_addr), Some(max_addr)) =
+                            (nat.min_addr.as_ref(), nat.max_addr.as_ref())
+                        {
+                            if min_addr != max_addr {
+                                nat_args.push_str(format!("={}-{}", min_addr, max_addr).as_str());
+                            } else {
+                                nat_args.push_str(format!("={}", min_addr).as_str());
+                            }
+                        }
+                    }
+                    if ct.flags & R_OVS_CT_NAT_RANGE_PROTO_SPECIFIED != 0 {
+                        nat_args.push_str(
+                            format!(
+                                ":{}-{}",
+                                nat.min_port.unwrap_or(0),
+                                nat.max_port.unwrap_or(0)
+                            )
+                            .as_str(),
+                        );
+                    }
+                    args.push(nat_args);
+                }
+                if ct.is_commit() {
+                    args.push("commit".to_string());
+                }
+                if ct.is_force() {
+                    args.push("force".to_string());
+                }
+                if ct.is_persistent() {
+                    args.push("persistent".to_string());
+                }
+                if ct.is_hash() {
+                    args.push("hash".to_string());
+                }
+                if ct.is_random() {
+                    args.push("random".to_string());
+                }
+                write!(f, " ct {}", args.join(","))?;
```

Comment:
I tried at first, but handing separators in a relatively long number of optional fields (essentially, implementing join) makes the the code uglier. But it would make it faster, so I'll give it another go